base/VulkanDevice.cpp

```diff
@@ -490,6 +490,9 @@ namespace vks
 		if (begin)
 		{
 			VkCommandBufferBeginInfo cmdBufInfo = vks::initializers::commandBufferBeginInfo();
+			// If we're allocating AND starting the command buffer, we're almost certainly going to 
+			// use it right away, so we'll use the ONE_TIME_SUBMIT_BIT flag
+			cmdBufInfo.flags |= VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT;
```

Comment:
I see no uses of  `vks::VulkanDevice::createCommandBuffer(..., true)` that aren't followed by some one off code and a vks::VulkanDevice::flushCommandBuffer` with the `free` parameter implicitly set tot true (I also see no uses of `vks::VulkanDevice::flushCommandBuffer` with flush set to false.  Have you considered a

```c++
vks::VulkanDevice::withOneTimeCommandBuffer(
    VkQueue, 
    VkCommandPool, 
    const std::function<void(VkCommandBuffer)>&);
```
function to encapsulate those `createCommandBuffer` / `flushCommandBuffer` pairs?

Reply:
Aside from trying to fix sync, my samples are in maintenance mode. I don't plan on putting a lot of work into them in the future. I mostly work on the Khronos samples.

Reply:
I'd like to contribute more to the official repo, but I find working with the C API excruciating after working with the C++ API for so long.